src/views/CreateEditHolder.js

```diff
@@ -104,73 +96,112 @@ export class EditArticle extends Component {
 
   handleInputChange = event => {
     const { name, value } = event.target;
-    const { onInputChange } = this.props;
     this.setState({ [name]: value });
-    onInputChange({ name, value });
   };
 
   submitArticle = () => {
-    const { article, editOneArticle, match } = this.props;
-    const { title, body, description, tagsList } = article;
-    const { slug } = match.params;
-    const error = Validator.newArticleValidation(article);
-    if (error && error.length) {
-      return this.setState({ error });
+    const {
+      editOneArticle,
+      createArticle,
+      history,
+      history: {
+        location: { pathname }
+      },
+      match: {
+        params: { slug }
+      }
+    } = this.props;
+    const { title, value, description } = this.state;
+
+    const error = Validator.newArticleValidation({
+      title,
+      body: value.toString("html"),
+      description
+    });
+    if (error) {
+      this.setState({ error });
+      return this.clearErrorMessage();
     }
     this.setState({
       error: ""
     });
-    editOneArticle({ title, body, description, tagsList }, slug);
+    if (pathname !== "/articles/new") {
+      return editOneArticle(
+        {
+          title,
+          body: value.toString("html"),
+          description
+        },
+        slug,
+        history
+      );
+    }
+    return createArticle(
+      {
+        title,
+        body: value.toString("html"),
+        description
+      },
+      history
+    );
   };
 
   handleNewTag = event => {
-    const { tag } = this.state;
+    const { tag, tagsList } = this.state;
     if (event.keyCode === 13 && tag.length) {
-      const { newTag } = this.props;
-      newTag({ tag });
-      this.setState({ tag: "" });
+      this.setState({ tag: "", tagsList: [...tagsList, tag] });
     }
   };
 
+  clearErrorMessage = () =>
```

Comment:
Is it possible to have those functions as utility functions?
Someone else may need them in the future.